staging/src/k8s.io/apiserver/pkg/endpoints/filters/warning.go

```diff
@@ -47,6 +72,11 @@ func (r *recorder) AddWarning(agent, text string) {
 	r.lock.Lock()
 	defer r.lock.Unlock()
 
+	// if we've already exceeded our limit and are already truncating, return early
+	if r.written >= truncateAtTotalRunes && r.truncating {
+		return
```

Comment:
utilruntime.HandleError() at least once so that we know this is happening server-side?  or a metric perhaps?

Reply:
logged an error once at the point where we flip to truncating